Branch-SDK/BNCSKAdNetwork.m

```diff
@@ -91,4 +101,94 @@ - (void)updatePostbackConversionValue:(NSInteger)conversionValue
     
 }
 
+- (void)updatePostbackConversionValue:(NSInteger)fineValue
+                          coarseValue:(NSString *)coarseValue
+                           lockWindow:(BOOL)lockWindow
+                    completionHandler:(void (^)(NSError *error))completion {
+    if (@available(iOS 16.1, *)) {
+        if ([self shouldAttemptSKAdNetworkCallout]) {
+            
+            ((id (*)(id, SEL, NSInteger, NSString *, BOOL, void (^)(NSError *error)))[self.skAdNetworkClass methodForSelector:self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow])(self.skAdNetworkClass, self.skAdNetworkUpdatePostbackConversionValueCoarseValueLockWindow, fineValue, coarseValue, lockWindow, completion);
+        }
+    }
+}
+
+- (int) calculateSKANWindowForTime:(NSDate *) currentTime{
+    
+    int firstWindowDuration = 2 * 24 * 3600;
+    int secondWindowDuration = 7 * 24 * 3600;
+    int thirdWindowDuration = 35 * 24 * 3600;
+    
+    NSTimeInterval timeDiff = [currentTime timeIntervalSinceDate:self.installDate];
+    
+    if (timeDiff <= firstWindowDuration) {
+        return 1;
+    } else if (timeDiff <= secondWindowDuration) {
+        return 2;
+    }else if (timeDiff <= thirdWindowDuration) {
+        return 3;
+    }
+    return 0;
```

Comment:
what is 0? we default to 0 for skanCurrentWindow, i would think 0 is the first window Duration and not 1, we can discuss this further during working session.

Reply:
@jf-branch Will discuss this during working session. Window 0 means There is no valid SKAN window, and it will help in initializing values.

Reply:
Maybe we should consider using an enum to make it a bit clearer that we mean 0 is no valid window.